bot_cli/commands.py

```diff
@@ -248,9 +270,76 @@ def birthdays(args):
 
     if get_birthdays_per_week:
         result = "Next week birthdays:\n" + "-" * 10 + "\n"
-        result += "\n".join([f"{day}: {celebrate_users}" for day, celebrate_users in get_birthdays_per_week.items()])
+        result += "\n".join(
+            [
+                f"{day}: {celebrate_users}"
+                for day, celebrate_users in get_birthdays_per_week.items()
+            ]
+        )
         result += "\n" + "-" * 10
         print(Fore.YELLOW + result)
     else:
-        print(Fore.YELLOW +
-              "There is no one to celebrate birthday next week")
+        print(Fore.YELLOW + "There is no one to celebrate birthday next week")
+
+
+@note_error_handler
+def add_note(notebook: Notes, args):
+    text = " ".join(args)
+    if text.isspace() or len(text) < MIN_NOTE_LEN:
+        raise ValueError(
+            f"Note cannot be empty and must be more than {MIN_NOTE_LEN} characters long"
+        )
+    note_id, _ = notebook.add_note(text)
+    print_success(f"Note with id {note_id} created successfully")
+
+
+# command, index, text = args[0], args[1], " ".join(args[2:])
+@note_error_handler
+def update_note(notebook: Notes, args):
+    index, text = args[0], " ".join(args[1:])
+    notebook.update_note(int(index), text)
+    print_success("Note successfully updated")
+
+
+@note_error_handler
+def replace_note(notebook: Notes, args):
+    index, text = args[0], " ".join(args[1:])
+    notebook.replace_note(int(index), text)
+    print_success("Note successfully replaced")
+
+
+@note_error_handler
+def remove_note(notebook: Notes, args):
+    index = args[0]
+    notebook.remove_note(int(index))
+    print_success("Note successfully removed")
+
+
+@note_error_handler
+def note_by_id(notebook: Notes, args):
+    index = args[0]
+    note = notebook.find_note_by_index(int(index))
+    if note["Tags"]:
+        str_tags = " ".join(note["Tags"])
+        print_success(f"Note: {note['Note']}\n Tags:{str_tags}")
+    else:
+        print_success(f"Note: {note['Note']}")
+
+
+@note_error_handler
+def note_by_text(notebook: Notes, args):
```

Comment:
The command should be `search-note` with search string and same name for the method

Reply:
done
cli - search-note
func -  search_note